Mathlib/Lean/Linter.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2025 Thomas R. Murrills. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas R. Murrills
+-/
+module
+
+public meta import Lean.Elab.Command
+public meta import Lean.Linter.Basic
+
+/-
+# Additional utilities for the `Linter` API
+-/
+
+public meta section
+
+open Lean Elab Command Linter
+
+namespace Lean.Linter
+
+variable {m : Type → Type} [Monad m] [MonadOptions m] [MonadEnv m]
+
+/--
+Runs a `CommandElabM` action when the provided linter option is `true`.
+
+Note: this definition is marked as `@[macro_inline]`, so it is okay to supply it with a linter option which has been registered in the same module.
+-/
+@[expose, macro_inline]
+def whenLinterOption (opt : Lean.Option Bool) (x : m Unit) : m Unit := do
+  if getLinterValue opt (← getLinterOptions) then x
```

Comment:
About naming: this is essentially an `if` statement so I was thinking something like:
```suggestion
def ifLinterOption (opt : Lean.Option Bool) (x : m Unit) : m Unit := do
  if getLinterValue opt (← getLinterOptions) then x
```